services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx

```diff
@@ -43,58 +48,126 @@ const activeFormPages = (draft: DraftSubmission): RouteT[] => {
     const [updateDraftSubmission, { error: updateError }] =
         useUpdateDraftSubmissionMutation()
 
     const updateDraft = async (
         input: UpdateDraftSubmissionInput
     ): Promise<DraftSubmission | undefined> => {
-        setShowFormAlert(false)
+        setShowPageErrorMessage(false)
         try {
             const updateResult = await updateDraftSubmission({
                 variables: {
                     input: input,
                 },
+                update(cache) {
+                        cache.evict({ id: `Submission2:${id}` })
+                        cache.gc()
+                 
+                },
             })
             const updatedSubmission: DraftSubmission | undefined =
                 updateResult?.data?.updateDraftSubmission.draftSubmission
 
-            if (!updatedSubmission) setShowFormAlert(true)
+            if (!updatedSubmission) setShowPageErrorMessage(true)
 
             return updatedSubmission
         } catch (serverError) {
-            setShowFormAlert(true)
+            setShowPageErrorMessage(true)
             return undefined
         }
     }
 
-    const draft = fetchData?.fetchDraftSubmission?.draftSubmission
-
     // Set up side effects
     useEffect(() => {
-        updateHeading(pathname, draft?.name)
-    }, [updateHeading, pathname, draft?.name])
+        if (formDataFromLatestRevision) {
+            const { name } = formDataFromLatestRevision
+            updateHeading(pathname, name)
+        }
+    }, [updateHeading, pathname, formDataFromLatestRevision])
+
+    useEffect(() => {
+        if (submissionAndRevisions) {
+            const currentRevisionPackageOrError =
+                getCurrentRevisionFromSubmission2(submissionAndRevisions)
+
+            // set form data
+            if (currentRevisionPackageOrError instanceof Error) {
+                setFormDataError('MALFORMATTED_DATA')
+            } else {
+                const formattedCurrentRevision = convertDomainModelFormDataToGQLSubmission(currentRevisionPackageOrError)
+                if (
+                    !formattedCurrentRevision
+                ) {
+                    setFormDataError('NOT_FOUND')
+                } else if (
+                    !isGQLDraftSubmission(formattedCurrentRevision)
+                ) {
+                    setFormDataError('WRONG_SUBMISSION_STATUS')
+                } else {
+                    // we think we have valid data
+                    setFormDataFromLatestRevision(formattedCurrentRevision)
+                }
+            } 
+
+            // set unlock info
+            if (submissionAndRevisions.status === 'UNLOCKED') {
+                const unlockedRevision = submissionAndRevisions.revisions.find(
+                    (rev) => rev.revision.unlockInfo
+                )
+                const unlockInfo = unlockedRevision?.revision.unlockInfo
+
+                if (
+                    unlockInfo?.updatedBy &&
```

Comment:
I’m surprised you can’t just do `if (unlockInfo)` here

Reply:
This is what the types look like in GQL

```
type UpdateInformation {
    updatedAt: Date
    updatedBy: String
    updatedReason: String
}
```

Is this what we intend 🔍 . Maybe those should all be required (with the `!`). Heres the way it looks in my editor on hover
![Screen Shot 2022-03-14 at 4 52 39 PM](https://user-images.githubusercontent.com/10750442/158267447-f61b21ac-c28e-481e-8603-f2a8cee962f2.png)

Reply:
Ahh, yeah, I think those should all be required. If we have an update info, it should have all the info in it.